sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -508,7 +563,13 @@ private LinkSubscription<AmqpSendLink> getSubscription(String linkName, String e
         org.apache.qpid.proton.amqp.transport.Target target, Map<Symbol, Object> linkProperties,
         AmqpRetryOptions options, TokenManager tokenManager) {
 
-        final Sender sender = session.sender(linkName);
+        final Sender sender;
+        try {
+            sender = protonSession.senderUnsafe(linkName);
+        } catch (ProtonSessionClosedException e) {
+            // The only time Exception can be thrown is in v2 mode with ProtonSession opted-in.
+            return new LinkSubscription<>(e);
```

Comment:
Earlier if the Qpid session gets closed after the invocation of "createProducer" (or "createConsumer") but before Qpid reactor thread runs "getSubscription" then the logic was to obtain new Sender (or Receiver) from closed Qpid session which relied on the undefined Qpid lib behavior to recover (whether link endpoint state emit error or complete in such case or timeout). We take care of such an edge case here by returning a retriable AmqpException (ProtonSessionClosedException).